src/main/java/com/acrolinx/sidebar/utils/MatchUtils.java

```diff
@@ -51,6 +52,23 @@ public static List<? extends AbstractMatch> filterDangerousMatches(
         .collect(Collectors.toList());
   }
 
+  public static List<? extends AbstractMatch> filterInsignificantWhitespaceMatches(
+      List<? extends AbstractMatch> abstractMatches, String lastCheckedContent) {
+    return abstractMatches.stream()
+        .filter(
+            match -> {
+              int startOffset = match.getRange().getMinimumInteger();
+              int endOffset = match.getRange().getMaximumInteger();
+              if ((endOffset - startOffset) > match.getContent().length()) {
+                String matchSurface = lastCheckedContent.substring(startOffset, endOffset);
+                final Pattern pattern = Pattern.compile("\\s{2,}", Pattern.DOTALL);
```

Comment:
Compiling a pattern is a more "expensive" step, thus one should avoid to do this more often if possible. In this case here, we can declare and compile the pattern before the stream() and only do it once.

Reply:
Fixed